cloudflare/resource_cloudflare_api_token.go

```diff
@@ -0,0 +1,248 @@
+package cloudflare
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/validation"
+	"log"
+	"strings"
+	"time"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceCloudflareApiToken() *schema.Resource {
+	p := schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"resources": {
+				Type:     schema.TypeMap,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"permission_groups": {
+				Type:     schema.TypeList,
+				Required: true,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+			},
+			"effect": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      "allow",
+				ValidateFunc: validation.StringInSlice([]string{"allow", "deny"}, false),
+			},
+		},
+	}
+
+	return &schema.Resource{
+		Create: resourceCloudflareApiTokenCreate,
+		Read:   resourceCloudflareApiTokenRead,
+		Update: resourceCloudflareApiTokenUpdate,
+		Delete: resourceCloudflareApiTokenDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"policy": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem:     &p,
+			},
+			"request_ip_in": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+			},
+			"request_ip_not_in": {
+				Type:     schema.TypeList,
+				Elem:     &schema.Schema{Type: schema.TypeString},
+				Optional: true,
+			},
+			"value": {
+				Type:      schema.TypeString,
+				Computed:  true,
+				Sensitive: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"issued_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"modified_on": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceCloudflareApiTokenCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*cloudflare.API)
+
+	name := d.Get("name").(string)
+
+	log.Printf("[INFO] Creating Cloudflare ApiToken: name %s", name)
+
+	t := cloudflare.APIToken{
+		Name:      name,
+		Policies:  resourceDataToApiTokenPolices(d),
+		Condition: resourceDataToApiTokenCondition(d),
+	}
+
+	t, err := client.CreateAPIToken(t)
+	if err != nil {
+		return fmt.Errorf("Error creating Cloudflare ApiToken %q: %s", name, err)
+	}
+
+	d.SetId(t.ID)
+	d.Set("status", t.Status)
+	d.Set("issued_on", t.IssuedOn.Format(time.RFC3339Nano))
```

Comment:
What is the thinking behind formatting as `RFC3339Nano` here when the timestamp is already `RFC3339`?